phpBB/phpbb/module/module_manager.php

```diff
@@ -0,0 +1,552 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\module;
+
+use phpbb\module\exception\module_not_found_exception;
+
+class module_manager
+{
+	/**
+	 * @var \phpbb\cache\driver\driver_interface
+	 */
+	protected $cache;
+
+	/**
+	 * @var \phpbb\db\driver\driver_interface
+	 */
+	protected $db;
+
+	/**
+	 * @var \phpbb\extension\manager
+	 */
+	protected $extension_manager;
+
+	/**
+	 * @var string
+	 */
+	protected $modules_table;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * @var string
+	 */
+	protected $php_ext;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\cache\driver\driver_interface	$cache				Cache driver
+	 * @param \phpbb\db\driver\driver_interface		$db					Database driver
+	 * @param \phpbb\extension\manager				$ext_manager		Extension manager
+	 * @param string								$modules_table		Module database table's name
+	 * @param string								$phpbb_root_path	Path to phpBB's root
+	 * @param string								$php_ext			Extension of PHP files
+	 */
+	public function __construct(\phpbb\cache\driver\driver_interface $cache, \phpbb\db\driver\driver_interface $db, \phpbb\extension\manager $ext_manager, $modules_table, $phpbb_root_path, $php_ext)
+	{
+		$this->cache				= $cache;
+		$this->db					= $db;
+		$this->extension_manager	= $ext_manager;
+		$this->modules_table		= $modules_table;
+		$this->phpbb_root_path		= $phpbb_root_path;
+		$this->php_ext				= $php_ext;
+	}
+
+	/**
+	 * Get row for specified module
+	 *
+	 * @param int		$module_id		ID of the module
+	 * @param string	$module_class	Class of the module (acp, ucp, mcp etc...)
+	 *
+	 * @return array
+	 *
+	 * @throws \phpbb\module\exception\module_not_found_exception	When there is no module with $module_id
+	 */
+	public function get_module_row($module_id, $module_class)
```

Comment:
Not sure about this, but the `$module_class` arg is there in every function, because that is how I replaced `$this->module_class`. I think it might be a good idea to get rid of it as `module_id` is a key by it self. Although if that extra check is needed for some reason, then that is that. So keep it or drop it?